### DISCUSSION TITLE
Shield Design V1

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#92] [Shield Design]
<img width="1330" alt="Screen Shot 2021-08-26 at 4 16 21 pm" src="https://user-images.githubusercontent.com/3038710/130923714-d838d41b-3066-44eb-9784-eef28d65b201.png">

I'm not too far from being done with the shield design. I've made some design choices with a view to keeping the design simple and accessible.

I'm aiming at a standard hammond case size with PCB dimensions of 100mm x 120mm

A 12v PSU will be required to power the system. The MAF needs 12V so it makes sense to use 12V as the system voltage. This is also good for the stepper motor which can utilise 12V for the 8825 stepper driver supply.

For the outputs (fan control) I'm using a [ULN2003A darlington array](https://www.ti.com/lit/ds/symlink/uln2003a.pdf?ts=1629939740539). These switch 12v to ground and can be used with inductive loads such as relays. This means you can use a regular automotive relay to switch your fans. NIce and accessible to everyone

I've decided to use a 4 channel 12 bit ADC to read the 5v pressure and MAF sensors. This uses I2C to communicate with the processor, which free's up 4 inputs. I've gone for an [ADS1015](https://www.adafruit.com/product/1083) on a breakout board (BOB). These are readily available and the BOB makes it easier for those with limited soldering skills to assemble the shield.

I've stuck with the GY-BME280 for baro, temp and relh measurement as this makes life very easy. Again for ease of assembly I've integrated it with it's BOB. I will also include some alternate headers as there are a few different flavours of these boards with different pin-outs.

I've decided to go with this modular approach rather than SMD or individual components so that the shield is easy to assemble. I've also doubled up on pressure sensor footprints so that the user can substitute the sensors for alternate package sizes if they wish. This goes hand in hand with the software design which should hopefully be easy to adapt for other  components

I'll post up a progress update when I have something more to report.

Happy to receive comments and suggestions...

**Comment by DeeEmm** [comment]
<img width="1304" alt="Screen Shot 2021-08-26 at 9 26 02 pm" src="https://user-images.githubusercontent.com/3038710/130958570-5c1db1ad-2d4a-40d0-b244-56f2d5d9c171.png">

Basic design is done. This is where I'm at.

- Two buck style LM2596 DC-DC converters (one for 5V one for 3.3V)
- BME280
- ADS1015 12Bit 4 Channel ADC to convert MAF, Ref pressure, Differential Pressure and Pitot sensor signals
- 8825 Stepper Motor driver
- ULN2003A Darlington array to switch 12v outputs (actually sinks external 12v signals)
- SMD and through-hole sensor connectors - use whichever you want.
- 2.1mm DC power jack
- Automotive blade fuse
- 2 x 16 pin Molex Mini-fit connectors
- Standard 100mm x 120mm board size to fit a Hammond 1455N1202 Case

All of these components are readily available, both online and via your local Jaycar / Maplins / Radio-Shack / etc

They all use common / universal footprints and so component brands should be interchangeable.

Apart from the pressure sensors, everything is through-hole and so should be easy-peasy to assemble

It will not be as cheap to build as directly mounting components to the PCB but the aim is try and strike a balance between cost and ease of assembly. Most of the sub components are SMD which puts assembly out of reach for many. So using break-out-boards and buck style PSUs helps to ensure that anyone can build it.

I still have network routing to do, but it's kinda getting there.

**Comment by DeeEmm** [comment]
![Layout Front](https://user-images.githubusercontent.com/3038710/131002267-d9d8df8f-eee4-40e8-aabc-57a2534da50b.png)

I've finished the PCB design and it looks okay. I'll get some made up and do some tests to validate.

I've uploaded the schematics and Gerbers to the [PCB folder](https://github.com/DeeEmm/DIY-Flow-Bench/tree/master/PCB) should someone want to make their own up but obviously these are untested at this stage.

I'll update this thread once I've received the boards

**Comment by DeeEmm** [comment]
I've added a Readme / BOM to the PCB folder and also updated the BOM page in the WIKI.

**Comment by DeeEmm** [comment]
Found a cheaper more accessible alternative to the molex headers...

![PP1154-16-pin-idc-locking-right-angle-headerImageMain-900](https://user-images.githubusercontent.com/3038710/131328567-96f665cc-5e0e-45b8-95f3-d241c2587510.jpg)

These use ribbon cable and have a standard pin spacing and so can be easily substituted with regular pin headers. The ribbon cable can be purchased by the metre and easily used to create a loom. The plugs also have the advantage that you do not need special tools to crimp the pins, unlike the molex, which can be pretty tricky.

![PS0985-16-way-idc-line-socketImageMain-900](https://user-images.githubusercontent.com/3038710/131328859-abe09363-cbb5-475c-8d4c-9ea7b0aa04fd.jpg)

Comes in fancy rainbow colours too

![WM4516-rainbow-cable-16-core-sold-per-metreImageMain-900](https://user-images.githubusercontent.com/3038710/131328879-304192fa-9ace-490d-ac2e-8c6bb480221b.jpg)

**Comment by DeeEmm** [comment]
So one thing that I have learned about ADC modules is that they can only measure up to VDD, meaning that if you use them on a 3.3v system they can only measure a maximum of 3.3v. This is an issue when using 5v FSD analog sensors and wanting to use the ADC as a way to translate between 5v and 3.3v devices. This is something that I initially missed.

The solution is to run the ADC on 5v and then use I2C logic converters to convert the I2C bus voltage down to 3.3v. Essentially you can connect all of the 3.3v I2C devices to the 3.3v side of the bus and all of the 5v I2C devices to the 5V side. It should be noted that each side of the bus requires its own pull-up resistors.

This is unfortunately a feature I omitted from the prototype. I will have to see if I can jumper one in.

**Comment by DeeEmm** [comment]
Code changes required for 1015 / 1115 ADC support have been added in merge request #95 

Note: FREQ based MAF needs to be attached direct to input as it needs pulse counter attached to it, which means that I need to add signal conditioning / level shifting to the MAF pin. If Voltage based MAF is used it attaches to ADC. This is because frequency measurement requires interrupt / pulse counter attached direct to pin.

So default method for reading 5v devices is that they are attached via the ADC. In turn the ADC connects via I2C to the ESP32, however. The ADC needs to have VDD set to 5V, which means that the I2C interface is also at 5v, which is too high for the ESP32. So the solution is to level shift the I2C bus using something like the PCA9306

**Comment by DeeEmm** [comment]
![IMG_20210903_143919](https://user-images.githubusercontent.com/3038710/133922482-57e8bf98-d6d8-48c3-bd36-64670b8195fd.jpg)

![IMG_20210903_143926](https://user-images.githubusercontent.com/3038710/133922503-b599dd03-3b62-4afd-a93f-ea157dc72cbf.jpg)

Initial shield testing is underway. The first prototype has been assembled and I am working through the initial setup. There are a few issues with this version so these corrections will be integrated into the next batch before the next order.

**Comment by bikedyno** [comment]
Hi ,have,you completed,  the shield design ?

**Comment by DeeEmm** [comment]
The shield is pretty much done but I've not been able to finish off testing to validate the changes. Unfortunately work has been hectic this past month or so so testing has ground to a halt. 

Once testing is complete I will publish the design and get some more boards made up, hopefully over the next few weeks

**Comment by bikedyno** [comment]
Cool , can't wait to see it

Thanks Pete

**Comment by markinferno** [comment]
@DeeEmm looking forward for that as well.

**Reply by pg7p** [reply]
Hello

How i can buy complete shield?

**Reply by DeeEmm** [reply]
At present I have not finished testing. Work / covid / other commitments have meant that this project has taken a bit of a back seat. I need to get it back to the top of the list again and get it finished off.

**Reply by pg7p** [reply]
Got it, I'll be waiting. Hopefully soon)
